### PR TITLE
DOC: linking interp1d docstring to tutorial

### LINKED ISSUE #18836: DOC: API reference doesn't point users to replacement for interp1d
### Issue with current documentation:

Looking at the [documentation for interp1d](https://docs.scipy.org/doc/scipy-1.11.1/reference/generated/scipy.interpolate.interp1d.html), I see it's legacy, but it wasn't clear what to replace it with. It took some searching before finding the explanation [in the user guide](https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html#legacy-interface-for-1-d-interpolation-interp1d).

### Idea or request for content:

Give some explanation or point to the user guide from the reference page.

### Additional context (e.g. screenshots, GIFs)

_No response_

### PR DESCRIPTION
**Pull request by RobPasMue**
#### Reference issue
Closes #18836

#### What does this implement/fix?
``interp1d`` docstring now links to the legacy ``interp1d`` tutorial.

#### Additional information
None.

### CODE REVIEW

**Comment by RobPasMue** [comment]
Pinging people involved in https://github.com/scipy/scipy/issues/18836 for discussion

@kjohnsen
@j-bowhay
@AlexSzatmary

**Comment by AlexSzatmary** [comment]
I would suggest instead something like, "See :ref:` <tutorial-interpolate_1D>`." to point to the new tutorial on the non-legacy tools. I think that should point to,
https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html#legacy-interface-for-1-d-interpolation-interp1d
which was referenced in the original issue.

**Comment by RobPasMue** [comment]
> I would suggest instead something like, "See :ref:` <tutorial-interpolate_1D>`." to point to the new tutorial on the non-legacy tools. I think that should point to, https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html#legacy-interface-for-1-d-interpolation-interp1d which was referenced in the original issue.

Hi @AlexSzatmary! I think the proper target is the one I mention in my PR (i.e. ``tutorial-interpolate_interp1d``). I'm happy to reword the sentence though to make it more simple if you all prefer.

File: scipy/interpolate/_interpolate.py [hunk 1]
```diff
@@ -391,7 +391,9 @@ class interp1d(_Interpolator1D):
 
     `x` and `y` are arrays of values used to approximate some function f:
     ``y = f(x)``. This class returns a function whose call method uses
-    interpolation to find the value of new points.
+    interpolation to find the value of new points. To find out more information
+    on how to use it, please refer to the
+    :ref:`legacy interp1d tutorial <tutorial-interpolate_interp1d>`.
```
Comment (j-bowhay) [inline scipy/interpolate/_interpolate.py:396]: I would add this under the legacy directive and phrase along the lines of "for a guide to the intended replacements to `interp1d` see ..."

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks for tackling this @RobPasMue, a suggestion

File: scipy/interpolate/_interpolate.py (hunk 1 above)
Comment (RobPasMue) [reply scipy/interpolate/_interpolate.py:396]: Sounds reasonable! Let me work on that.

**Comment by j-bowhay** [comment]
Thanks @RobPasMue! Would you mind pushing that last commit with just `[skip actions] [skip cirrus]`, I would like to make sure the doc build is all good

**Comment by RobPasMue** [comment]
> Thanks @RobPasMue! Would you mind pushing that last commit with just `[skip actions] [skip cirrus]`, I would like to make sure the doc build is all good

Right away!

**Comment by RobPasMue** [comment]
FYI @j-bowhay  - tried building it locally, it worked fine

image

**Comment by AlexSzatmary** [comment]
In looking over my previous comment, I see that I had miscommunicated a bit. I used the URL requested by the OP,
https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html#legacy-interface-for-1-d-interpolation-interp1d
which is what is referred to in the proposed change. However, that points to the legacy interface. Because the goal is to point people to the replacement, I'd suggest instead <tutorial-interpolate_1D>, which I think points to,
https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html
It's not a huge difference because it's ultimately the same page.

**Comment by j-bowhay** [comment]
Thanks for pointing that out @AlexSzatmary, @RobPasMue if you could update the link then this should be good to go

**Comment by RobPasMue** [comment]
Hi @j-bowhay @AlexSzatmary - the reference is pointing at the location right where the Legacy interp1d tutorial starts. See image below

image


Meaning that when you click on the link you get redirected to that exact location. See video:

[video.webm](https://github.com/scipy/scipy/assets/37798125/d35213f4-d046-49d2-a23f-f07e9ac12c3d)

**Comment by RobPasMue** [comment]
Effectively, the code is behaving as requested. I'd avoid explicitly pointing to external links and using internal Sphinx references. I think that the only way to use explicitly the ``URL#Section`` approach is by using external links.

**Comment by j-bowhay** [comment]
To clarify, the original issue is asking to be directed on where to find info on `interp1d` replacements not on `interp1d` itself hence the request to go to the top of the `1-D interpolation` tutorial, not the Legacy interp1d section. For example, if you were using `kind='linear'` you would want to look at https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html#piecewise-linear-interpolation. Hope that makes sense?

**Comment by AlexSzatmary** [comment]
Hi @RobPasMue I agree that what you did is the correct way of pointing to the link that was requested by the OP. However, I instead suggest using the reference <tutorial-interpolate_1D>, which points to the top part of the 1D interpolation tutorial so that users know how to use the non-legacy interface. OP's issue was that they saw a legacy warning but didn't know how to do things differently. I think that OP requested a link to the right doc but their URL (mistakenly?) pointed to the interp1d part and I don't think that was their intention.

**Comment by RobPasMue** [comment]
Oh okay, got it! Sorry I misunderstod it. Changing it right away

**Comment by RobPasMue** [comment]
The link to the top part of the page was ``tutorial-interpolate_1Dsection``, just for the sake of completeness.

**Comment by RobPasMue** [comment]
> To clarify, the original issue is asking to be directed on where to find info on `interp1d` replacements not on `interp1d` itself hence the request to go to the top of the `1-D interpolation` tutorial, not the Legacy interp1d section. For example, if you were using `kind='linear'` you would want to look at https://docs.scipy.org/doc/scipy-1.11.1/tutorial/interpolate/1D.html#piecewise-linear-interpolation. Hope that makes sense?

Makes total sense now, indeed! I'm new ti SciPy and I wanted to contribute the best way I could... Looks like I got it wrong a couple of times haha! I hope now everything is fine. We are pointing to the top of the 1D interpolation tutorial page.

Thanks for all your support!

**Comment by AlexSzatmary** [comment]
It might be nice if at some point a full replacement for interp1d were developed (doing nearest, next, and previous). I think that's out of scope for this issue.

@RobPasMue I'm also wanting to get into SciPy development and I feel like I had my share of confusion, too! I'd have taken care of this myself but I'm still setting up the dev environment.

**Review by j-bowhay — APPROVED** [review]
Thanks @RobPasMue, keep them coming!

**Comment by j-bowhay** [comment]
@RobPasMue @AlexSzatmary it is great that you are both interested in SciPy development! We have a fortnightly community call, monthly new contributors meeting and a slack channel. Feel free to use any/all of these to get the help you require. We are aware things can be confusing to start so feel free to reach out for help. https://scipy.org/community/

**Pull request merged**